data/blog/2024-04-26-goodbye-connectors.mdx

```diff
@@ -0,0 +1,29 @@
+
+**Failure Log Management**: Providing failure logs to our users proved to be a challenging endeavor. The framework lacked adequate mechanisms to separate user logs, making it impossible for us to furnish users with relevant failure logs. Despite Kafka Connect offering a way to report failures, not all Connector implementations does it correctly. Sometimes connector looks working, but we can see it only failed on the logs, or eventough it reports that it is not working, the report does not contain the exact reason of the failure. These inconsistencies in Connector implementations often resulted in manual efforts to sift through logs, an inherently unscalable approach.
+
+**Version Upgrade Complexities**: The saga of version upgrades is a tale as old as time. Users would report bugs in specific Connector versions, prompting fixes that were sometimes only available in non-backward compatible updates. Balancing the needs of different users while ensuring compatibility across Connector versions proved to be a Herculean task. The cost and effort associated with serving every requested Connector version is not feasible, both in terms of infrastructure and maintenance.
+
+**Dependency on Kafka Versions**: To add insult to injury, certain Connector implementations couldn't seamlessly transition to newer Kafka versions, effectively handcuffing us to older versions. This dependency limited our ability to upgrade Kafka brokers, hindering our mission to deliver the best possible experience for our users.
+
+Despite these challenges, our commitment to providing you with the best possible Kafka experience remains unwavering. While Kafka Connect serverless may be stepping out of the limelight, Kafka itself—and its trusty companion, Schema Registry—remain steadfast pillars of our ecosystem.
+
+So what's the plan now, you ask? Well, we're doubling down on what we do best: providing you with robust Kafka infrastructure. And don't worry, we're not leaving you high and dry. We're arming you with [all the knowledge and resource](PROVIDE LINK HERE) you need to roll your own Kafka Connect framework, whether you're a server guru or a cloud enthusiast. Furthermore, while you won't be able to create new connectors moving forward, we want to assure you that your existing connectors will continue to run smoothly for the next 6 months, until September 1, 2024. This grace period should allow ample time for any necessary adjustments or transitions. So rest assured, we've got your back every step of the way. And remember, should you have any questions or need further assistance, feel free to reach out to us at [support@upstash.com](mailto::support@upstash.com) or join our community on [Discord](https://discord.gg/w9SenAtbme). We're here to help!
```

Comment:
Let's change the discord url to `https://upstash.com/discord`, this static url may be broken after a while. We keep the /discord url updated